tools/linter/mypy/requirements.txt

```diff
@@ -0,0 +1,5 @@
+numpy==1.20
+expecttest==0.1.3
+mypy==0.812
+junitparser
+rich
```

Comment:
we should pin these too (whatever is the latest release right now is probably fine)